### PR TITLE
Chore: Adds basic structure for e2e tests (GLVSC-578)

### PR DESCRIPTION
**Pull request by sergiolms**
# Description

This PR adds the base structure from where we can build E2E tests.
For it, it's using [wdio](https://webdriver.io/docs/wdio-vscode-service/) as framework.
Some features to highlight:
- In `wdio.conf.ts` we find the `capabilities` key, where we can specify more than one browser (by default set to `vscode`). This would allow us to test GitLens in different browsers such as Chrome or Firefox.
- The new launch action `Watch & Run E2E Tests` also passes a debugging flag, that allow developers to add breakpoints and debug the tests when creating new.
- By pointing in the conf file to the root folder, the extension is automatically installed so all tests start with a fresh install (without needing to build a `.vsix` file).
- In `wdio.conf.ts` file, we can find extra parameters to mocha. This one:
   ```
      mochaOpts: {
		timeout: 10000,
      },
   ```
   Refers to the time mocha waits until a test responds. Setting this to high values or `Infinity` will help developers to debug. Otherwise it could lead to early exiting while debugging a test.
- Sample tests were added to verify the extension was successfully installed.

# Checklist

- [x] I have followed the guidelines in the Contributing document
- [x] My changes follow the coding style of this project
- [x] My changes build without any errors or warnings
- [x] My changes have been formatted and linted
- [x] My changes have been rebased and squashed to the minimal number (typically 1) of relevant commits
- [x] My changes have a descriptive commit message with a short title, including a `Fixes $XXX -` or `Closes #XXX -` prefix to auto-close the issue that your PR addresses

### CODE REVIEW

File: tests/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		"module": "ESNext",
+		"target": "es2022",
+		"lib": ["es2022", "dom"],
+		"types": [
+			"node",
+			"@wdio/globals/types",
+			"@wdio/mocha-framework",
+			"webdriverio/async",
+			"expect-webdriverio",
+			"webdriverio/async",
```
Comment (nzaytsev) [inline tests/tsconfig.json:13]: duplicated item

File: tests/specs/test.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { browser, expect } from '@wdio/globals';
+import path from 'path';
+import url from 'url';
+
+const __dirname = url.fileURLToPath(new URL('.', import.meta.url));
+
+describe('VS Code Extension Testing', () => {
+	let workbench: Workbench;
```
Comment (nzaytsev) [inline tests/specs/test.e2e.ts:8]: seems to be untyped now

File: tests/tsconfig.json (hunk 1 above)
Comment (nzaytsev) [reply tests/tsconfig.json:13]: also seems to be redundant
https://stackoverflow.com/questions/75057693/cannot-find-type-definition-file-for-wdio-globals-types

File: tests/helpers/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { browser } from '@wdio/globals';
+import type { EditorView, Workbench } from 'wdio-vscode-service';
+import type { ExtensionPageHandler } from '../types/helpers.ts';
+
+export async function waitUntilExtensionLoaded(): Promise<ExtensionPageHandler> {
+	const welcomePageTitle = 'Welcome to GitLens';
```
Comment (nzaytsev) [inline tests/helpers/helpers.ts:6]: could be fine to reuse from code 
https://github.com/gitkraken/vscode-gitlens/blob/c3a35b4362f153454541a4aea435b3d2f9f05bac/src/webviews/welcome/registration.ts#L13

File: tests/helpers/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+	const wb: Workbench = await browser.getWorkbench();
+	const editorView: EditorView = wb.getEditorView();
```
Comment (nzaytsev) [inline tests/helpers/helpers.ts:8]: redundant typing

File: tests/helpers/helpers.ts (hunk 3 above)
Comment (sergiolms) [reply tests/helpers/helpers.ts]: Tried this but I'm finding some problems with ES and JS configuration. Let's add this as an improvement, shall we?

**Comment by sergiolms** [comment]
Converting this into a draft for now. We may want to explore Playwright first before going with Webdriverio

**Comment by eamodio** [comment]
Are we moving forward with playwright or wdio on this one? I think it was playwright, if so, can we remove the wdio (or maybe better yet, leave this PR, so we have its history), and open a new on for playwright only.

**Comment by sergiolms** [comment]
> Are we moving forward with playwright or wdio on this one? I think it was playwright, if so, can we remove the wdio (or maybe better yet, leave this PR, so we have its history), and open a new on for playwright only.

Absolutely, let me close this PR and open a new one with just the Playwright implementation 😄

**Pull request closed without merging**